packages/create-amplify/src/package-manager-controller/package_manager_controller.ts

```diff
@@ -0,0 +1,121 @@
+import { existsSync as _existsSync } from 'fs';
+import _fsp from 'fs/promises';
+import { execa as _execa } from 'execa';
+import * as _path from 'path';
+import { logger } from '../logger.js';
+import { executeWithDebugLogger as _executeWithDebugLogger } from '../execute_with_logger.js';
+import { DependencyType } from './package_manager_controller_factory.js';
+
+/**
+ * PackageManagerController is an abstraction around package manager commands that are needed to initialize a project and install dependencies
+ */
+export abstract class PackageManagerController {
+  /**
+   * constructor - sets the project root
+   */
+  constructor(
+    readonly projectRoot: string,
+    protected executable: string,
+    protected binaryRunner: string,
+    protected initDefault: string[],
+    protected installCommand: string,
```

Comment:
these can be read only (here and there if this repeats)

Reply:
Oops...just merged. I'll address it in the next PR

Reply:
Followed up here https://github.com/aws-amplify/amplify-backend/pull/901#discussion_r1456624715